src/ksm/group.go

```diff
@@ -4,52 +4,52 @@ import (
 	"fmt"
 
 	"github.com/newrelic/infra-integrations-sdk/log"
+	corev1 "k8s.io/api/core/v1"
 
-	"github.com/newrelic/nri-kubernetes/v2/src/client"
 	"github.com/newrelic/nri-kubernetes/v2/src/data"
 	"github.com/newrelic/nri-kubernetes/v2/src/definition"
-	"github.com/newrelic/nri-kubernetes/v2/src/ksm/metric"
 	"github.com/newrelic/nri-kubernetes/v2/src/prometheus"
 )
 
 type ksmGrouper struct {
-	queries   []prometheus.Query
-	client    client.HTTPGetter
-	logger    log.Logger
-	k8sClient client.Kubernetes
+	queries                []prometheus.Query
+	filteredMetricFamilies prometheus.FilteredMetricFamilies
+	logger                 log.Logger
+	services               []*corev1.Service
 }
 
-// addServiceSpecSelectorToGroup adds a new metric to the service group
-// which includes the selectors defined in the service spec.
-func (r *ksmGrouper) addServiceSpecSelectorToGroup(serviceGroup map[string]definition.RawMetrics) error {
-	services, err := r.k8sClient.ListServices("")
-	if err != nil {
-		return fmt.Errorf("listing services: %w", err)
-	}
-	for _, s := range services.Items {
-		serviceRawMetrics, ok := serviceGroup[fmt.Sprintf("%s_%s", s.Namespace, s.Name)]
-		if !ok {
-			continue
-		}
+type GrouperConfig struct {
+	Queries              []prometheus.Query
+	MetricFamiliesGetter prometheus.FilteredMetricFamilies
+	Logger               log.Logger
+	Services             []*corev1.Service
```

Comment:
I think we intend for this to be passed down to the `ksmGrouper.services` but we never do that. Is this correct?